packages/admin/cms-admin/src/generator/future/generateGrid.ts

```diff
@@ -321,24 +321,30 @@ export function generateGrid(
 
         let renderCell: string | undefined = undefined;
         let valueGetter: string | undefined = name.includes(".") ? `({ row }) => row.${name.replace(/\./g, "?.")}` : undefined;
+        let valueFormatter: string | undefined = undefined;
 
         let gridType: "number" | "boolean" | "dateTime" | "date" | undefined;
 
         if (type == "dateTime") {
             valueGetter = `({ row }) => row.${name} && new Date(row.${name})`;
+            valueFormatter = `({ value }) => value ? intl.formatDate(value, { day: "numeric", month: "numeric", year: "numeric", hour: "numeric", minute: "numeric" }) : ""`;
             gridType = "dateTime";
         } else if (type == "date") {
             valueGetter = `({ row }) => row.${name} && new Date(row.${name})`;
+            valueFormatter = `({ value }) => value ? intl.formatDate(value) : ""`;
             gridType = "date";
         } else if (type == "number") {
             gridType = "number";
+            valueFormatter = `({ value }) => typeof value === "number" ? value.toString() : ""`;
```

Comment:
what happens if you don't have these valueFormatters?

Reply:
Since the value is not a string, the following error will be thrown:

https://github.com/vivid-planet/comet/blob/3ea66fb38716e95a833d8126510807989d0c9b08/packages/admin/admin/src/dataGrid/excelExport/generateExcelFile.ts#L55-L57

Reply:
Actually, since `number` and `null` are explicitly allowed in the rows type it probably makes sense to allow those values as well instead of throwing an error. 

In that case the `valueFormatter` would not be necessary, unless the number should be formatted in some way, which is currently not possible (as far as I know). 

Should I add support for these types and remove the `valueFormatter` (at least until we support formatting the numbers)?

https://github.com/vivid-planet/comet/blob/3ea66fb38716e95a833d8126510807989d0c9b08/packages/admin/admin/src/dataGrid/excelExport/generateExcelFile.ts#L41

Reply:
> Should I add support for these types and remove the valueFormatter

yes, please

Reply:
Done in 68a58a1ef0521cc7f59f0fad46a4808ec48d7025